plugins/outputs/amqp/amqp.go

```diff
@@ -296,22 +296,22 @@ func (q *AMQP) publish(key string, body []byte) error {
 func (q *AMQP) serialize(metrics []telegraf.Metric) ([]byte, error) {
 	if q.UseBatchFormat {
 		return q.serializer.SerializeBatch(metrics)
-	} else {
-		var buf bytes.Buffer
-		for _, metric := range metrics {
-			octets, err := q.serializer.Serialize(metric)
-			if err != nil {
-				log.Printf("D! [outputs.amqp] Could not serialize metric: %v", err)
-				continue
-			}
-			_, err = buf.Write(octets)
-			if err != nil {
-				return nil, err
-			}
+	}
+
+	var buf bytes.Buffer
+	for _, metric := range metrics {
+		octets, err := q.serializer.Serialize(metric)
+		if err != nil {
+			log.Printf("D! [outputs.amqp] Could not serialize metric: %v", err)
```

Comment:
That output should really use the telegraf logger, but that's probably another issue. Should we open one?

Reply:
No, I will change it here :)

Reply:
"log" packages removed from files which were changed in previous commit. I replaced it by `telegraf.Logger` when it was easy, hard cases need to wait...